### ISSUE TITLE
Feature Request: change DuetRRF Connection on machine change

### ISSUE THREAD

**Issue by cbusillo** [Kriechi/Cura-DuetRRFPlugin#8]
Hi!  Is it possible to have the duet connection change when the printer is changed?  I'm running into the issue where I click print and forget to change the output.  Maybe a way to connect the connection to the machine?

**Comment by Kriechi** [comment]
I'm sorry - I don't fully understand what you want to change?
Can you explain what you mean by "duet connection change when the printer is changed"?

**Comment by cbusillo** [comment]
Sorry this is really hard for me to explain.  I took screen shots that might explain it better.  Basically I want the Duet Connection tied to a particular machine in Cura.
<img width="461" alt="screen shot 2018-03-31 at 9 35 59 am" src="https://user-images.githubusercontent.com/1875516/38163760-679c412a-34c7-11e8-9484-c12e7e7ec136.png">
<img width="419" alt="screen shot 2018-03-31 at 9 36 38 am" src="https://user-images.githubusercontent.com/1875516/38163762-67baa958-34c7-11e8-93cb-0a9b72ebdd50.png">
<img width="460" alt="screen shot 2018-03-31 at 9 36 25 am" src="https://user-images.githubusercontent.com/1875516/38163765-7af36a5a-34c7-11e8-8c7b-fcc217821abf.png">

**Comment by Kriechi** [comment]
Ok - I think I get the idea of what you want.
I need to find time to look into the Cura plugin system - as there is no proper documentation this might be either really difficult to implement - or it might be really easy. 

If anyone wants to contribute and send a PR that implements this feature - I'm happy to merge it!

**Comment by GT1ZA** [comment]
Hi, im using cura 4.5 and cant seem to find this in either the marketplace or get the extension to appear with manual installation. Is it working in 4.5?

**Comment by Kriechi** [comment]
@GT1ZA please open a new ticket in the future for new questions and problems.
The latest Cura version 4.6 is supported by the Cura-DuetRRF plugin.

**Comment by Kriechi** [comment]
@cbusillo I just pushed a new branch with a complete refactoring of the plugin settings - and I think this should address your feature request of having the settings attached / switched with the Cura printers.

Are you comfortable installing the plugin "from source" into Cura from this branch?
https://github.com/Kriechi/Cura-DuetRRFPlugin/tree/new-settings
Basically download the [zip archive](https://github.com/Kriechi/Cura-DuetRRFPlugin/archive/new-settings.zip) and extract it into `~/Library/Application Support/Cura/4.7/plugins/`

**Comment by cbusillo** [comment]
Hi! I no longer have the printers. I think others will appreciate this feature though! Would you like me to close this?

**Comment by ericlmccormick** [comment]
This is greatly appreciated. Thank you!

**Comment by Kriechi** [comment]
@ericlmccormick did you test it? can you report it is working an does what you expected it to do?
I'm not sure yet how to handle existing user data and printer configurations. Migrating them over to the new settings and matching them against Cura printers is difficult and error prone...

**Comment by ericlmccormick** [comment]
Yes. I have two printers running duet boards. I had been using PrusaSlicer for a while. I think when I last used Cura I was printing via octaprint. I noticed that my start and end codes were not working using the plug in from the cura market. I think your plug in sent the code generated, ignoring the machine setup into which includes the start and stop gcode. I might be wrong and may have messed something up.
However, I manually installed the plug-in from here and was able to link the machine profiles to their duet addresses similar to the octoprint plugin. Now when I select a machine, it grabs the correct profiles and sends to the correct machine. This really helps the work flow and grabbed the start and end gcode commands.
